tests/utils/test_annotations.py

```diff
@@ -133,3 +185,125 @@ def test_deprecated_and_keyword_second():
+    with pytest.warns(FutureWarning, match=re.escape(msg)):
+        DeprecatedDataClass(x=10, y=20)
+    assert msg in str(DeprecatedDataClass.__doc__)
+
+
+def test_deprecated_dataclass_fields():
+    msg = "``tests.utils.test_annotations.DeprecatedDataClass`` is deprecated since 1.0.0"
+    with pytest.warns(FutureWarning, match=re.escape(msg)):
+        instance = DeprecatedDataClass(x=5, y=15)
+        assert instance.x == 5
+        assert instance.y == 15
+    assert msg in str(DeprecatedDataClass.__doc__)
+
+
+def test_deprecated_dataclass_method():
+    msg = "``tests.utils.test_annotations.AnotherDeprecatedDataClass`` is deprecated since 1.0.0"
+    with pytest.warns(FutureWarning, match=re.escape(msg)):
+        instance = AnotherDeprecatedDataClass(a=3, b=4)
+        assert instance.add() == 7
+    assert msg in str(AnotherDeprecatedDataClass.__doc__)
+
+
+def test_deprecated_dataclass_different_order():
+    msg = "``tests.utils.test_annotations.AnotherDeprecatedDataClassOrder`` is deprecated"
+    with pytest.warns(FutureWarning, match=re.escape(msg)):
+        AnotherDeprecatedDataClassOrder(m=7, n=8)
+    assert msg in str(AnotherDeprecatedDataClassOrder.__doc__)
+
+
+def test_deprecated_dataclass_dunder_methods():
+    instance = DeprecatedDataClass(x=1, y=2)
+
+    assert instance.x == 1
+    assert instance.y == 2
+
+    expected_repr = "DeprecatedDataClass(x=1, y=2)"
+    assert repr(instance) == expected_repr
+
+    instance2 = DeprecatedDataClass(x=1, y=2)
+    instance3 = DeprecatedDataClass(x=2, y=3)
+    assert instance == instance2
+    assert instance != instance3
+
+
+def test_deprecated_dataclass_preserves_fields():
+    instance = DeprecatedDataClass(x=100, y=200)
+    field_names = {f.name for f in fields(DeprecatedDataClass)}
+    assert field_names == {"x", "y"}
+    assert instance.x == 100
+    assert instance.y == 200
+
+
+def test_deprecated_dataclass_preserves_methods():
+    instance = DeprecatedDataClass(x=10, y=20)
+    assert instance.add() == 30
+
+
+def test_deprecated_dataclass_preserves_class_attributes():
+    assert DeprecatedDataClass.__module__ == "tests.utils.test_annotations"
+    assert DeprecatedDataClass.__qualname__ == "DeprecatedDataClass"
+
+
+def test_deprecated_dataclass_dunder_methods_not_mutated():
+    instance = DeprecatedDataClass(x=5, y=10)
+    assert instance.x == 5
+    assert instance.y == 10
+
+    expected_repr = "DeprecatedDataClass(x=5, y=10)"
+    assert repr(instance) == expected_repr
+
+    same_instance = DeprecatedDataClass(x=5, y=10)
+    different_instance = DeprecatedDataClass(x=1, y=2)
+    assert instance == same_instance
+    assert instance != different_instance
+
+    assert instance.add() == 15
+
+    allowed_attrs = {"x", "y", "add", "__module__", "__doc__", "__annotations__"}
+    for attr in dir(instance):
+        if not attr.startswith("__"):
+            assert attr in allowed_attrs
```

Comment:
What does this part do?

Reply:
The original implementation of the decorator was mutating the dunder commands when applied to classes and dataclasses since we were applying the patch to the top-level object reference (weird) so this test is just making sure that we're not mutating anything or dropping any dunder commands that are set by dataclass definitions